pytorch_lightning/overrides/data_parallel.py

```diff
@@ -63,7 +65,39 @@ def forward(self, *inputs, **kwargs):
 
         replicas = self.replicate(self.module, self.device_ids[:len(inputs)])
         outputs = self.parallel_apply(replicas, inputs, kwargs)
-        return self.gather(outputs, self.output_device)
+        return self.gather(outputs)
+
+    def gather(self, outputs):
```

Comment:
do you think we could reuse the implementation in the `apply_to_collection` here? This looks very familiar to the recursive traversal in nested objects that we have elsewhere in Lightning (device transfers, metrics, ...). I would prefer to reduce duplicate code as much as possible :)

Reply:
Sounds neat ;) I'll have a look.

Reply:
I had no luck using `apply_to_collection `. It has a different approach and can't be used here. But I made some edits so the codes are now more similar.